app/routes/user/endorsements.server.ts

```diff
@@ -0,0 +1,292 @@
+            }
+          : {
+              ':endorserSub': this.#sub,
+              ':status': 'pending',
+            },
+      ProjectionExpression:
+        '#requestorSub, #requestorEmail, #endorserSub, #endorserEmail, #status, #created',
+    }
+
+    const db = await tables()
+    const result = await db.circular_endorsements.query(queryParams)
+    return result.Items as EndorsementRequest[]
+  }
+
+  /**
+   * Checks that the provided endorserSub is valid to be used for a new request.
+   *
+   * Throws an HTTP error if:
+   *  - The endorserSub is the current user's sub
+   *  - The endorserSub does not correspond to an existing user
+   *  - The user exists, but does not belong to the submitters group
+   *
+   * @param endorserSub - the sub of the user who will approve or reject the request
+   * @returns true if all checks succeed
+   */
+  async requestedEndorserIsValid(endorserSub: string): Promise<boolean> {
+    if (endorserSub === this.#sub)
+      throw new Response(
+        'Users cannot request themselves as the endorser of their own request',
+        {
+          status: 400,
+        }
+      )
+
+    const escapedEndorserString = endorserSub.replaceAll('"', '\\"')
+
+    const user = (
+      await cognitoIdentityProviderClient.send(
+        new ListUsersCommand({
+          UserPoolId: process.env.COGNITO_USER_POOL_ID,
+          Filter: `sub = "${escapedEndorserString}"`,
+        })
+      )
+    )?.Users?.[0]
+    if (!user)
+      throw new Response('Requested user does not exist', {
+        status: 400,
+      })
+
+    const getGroupsCommand = new AdminListGroupsForUserCommand({
+      Username: user.Username,
+      UserPoolId: process.env.COGNITO_USER_POOL_ID,
+    })
+    const groupsResponse = await cognitoIdentityProviderClient.send(
+      getGroupsCommand
+    )
+
+    if (
+      !groupsResponse.Groups ||
+      !groupsResponse.Groups.find(
+        (x) => x.GroupName == 'gcn.nasa.gov/circular-submitter'
+      )
+    )
+      throw new Response('User is not in the submitters group', {
+        statusText: 'Bad Request',
+        status: 400,
+      })
+
+    return true
+  }
+
+  /**
+   * This should be used in future updates to create a list of valid users for
+   * the current user to select from when creating a new endorsement request.
+   *
+   * @returns a list of users in the circular-submitter group.
+   */
+  async getSubmitterUsersFromCognito() {
+    const command = new ListUsersInGroupCommand({
+      GroupName: 'gcn.nasa.gov/circular-submitter',
+      UserPoolId: process.env.COGNITO_USER_POOL_ID,
+    })
+    const result = await cognitoIdentityProviderClient.send(command)
+    return result.Users
+  }
+
+  async #buildNewRequestFromEndorserSub(
+    endorserSub: string
+  ): Promise<EndorsementRequest> {
+    const getEndorserDataCommand = new ListUsersCommand({
+      UserPoolId: process.env.COGNITO_USER_POOL_ID,
+      Filter: `sub = "${endorserSub}"`,
```

Comment:
endorserSub must be escaped. Isn't this code duplicated?

Reply:
will be handled in related change from one of your comments above